internal/builders/container/README.md

```diff
@@ -152,6 +153,125 @@ jobs:
+  # EXAMPLE:
+  # projects/123123412578/locations/global/workloadIdentityPools/my-pool/providers/my-provider
+  WORKLOAD_IDENTITY_PROVIDER: ${{ vars.WORKLOAD_IDENTITY_PROVIDER }}
+  # EXAMPLE:
+  # my-service-account@blank-check-231234.iam.gserviceaccount.com
+  SERVICE_ACCOUNT: ${{ vars.SERVICE_ACCOUNT }}
+
+on: [push]
+
+jobs:
+  # This step builds our image, pushes it, and outputs the repo hash digest.
+  build:
+    permissions:
+      contents: read
+      packages: write
+      id-token: write
+    outputs:
+      image: ${{ steps.image.outputs.image }}
+      digest: ${{ steps.build.outputs.digest }}
+      workload_identity_provider: ${{ steps.idprov.outputs.widp }}
+      service_account: ${{ steps.sa.outputs.sa }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # v2.3.4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@dc7b9719a96d48369863986a06765841d7ea23f6 # v2.0.0
+
+      - id: 'auth'
+        name: 'Authenticate to Google Cloud'
+        uses: 'google-github-actions/auth@v1.0.0'
+        with:
+          token_format: 'access_token'
+          workload_identity_provider: ${{ vars.WORKLOAD_IDENTITY_PROVIDER }}
+          service_account: ${{ vars.SERVICE_ACCOUNT }}
+
+      - name: Authenticate Docker
+        uses: docker/login-action@49ed152c8eca782a232dede0303416e8f356c37b # v2.0.0
+        with:
+          registry: northamerica-northeast1-docker.pkg.dev
+          username: oauth2accesstoken
+          password: ${{ steps.auth.outputs.access_token }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@69f6fc9d46f2f8bf0d5491e4aabe0bb8c6a4678a # v4.0.1
+        with:
+          images: ${{ vars.REPOSITORY_PATH }}/${{ env.IMAGE_NAME }}
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@e551b19e49efd4e98792db7592c17c09b89db8d8 # v3.0.0
+        id: build
+        with:
+          push: true
+          tags: |
+            ${{ steps.meta.outputs.tags }}
+            ${{ vars.REPOSITORY_PATH }}/${{ env.IMAGE_NAME }}:latest
+          labels: ${{ steps.meta.outputs.labels }}
+
+      - name: Output image
+        id: image
+        run: |
+          # NOTE: Set the image as an output because the `env` context is not
+          # available to the inputs of a reusable workflow call.
+          image_name="${REPOSITORY_PATH}/${IMAGE_NAME}"
+          echo image=$image_name >> "$GITHUB_OUTPUT"
+
+      - name: Output workload_identity_provider
+        id: idprov
+        run: |
+          workload_identity_provider=${WORKLOAD_IDENTITY_PROVIDER}
+          echo widp=$workload_identity_provider >> "$GITHUB_OUTPUT"
+
+      - name: Output service_account
+        id: sa
+        run: |
+          service_account=${SERVICE_ACCOUNT}
+          echo sa=$service_account >> "$GITHUB_OUTPUT"
```

Comment:
nit: You could combine these into one step and save yourself some lines of code. This is often helpful in docs.

```suggestion
      - name: Outputs
        id: outputs
        run: |
          # NOTE: Set the outputs because the `env` context is not
          # available to the inputs of a reusable workflow call.
          image_name="${REPOSITORY_PATH}/${IMAGE_NAME}"
          echo image=$image_name >> "$GITHUB_OUTPUT"
          
          workload_identity_provider=${WORKLOAD_IDENTITY_PROVIDER}
          echo widp=$workload_identity_provider >> "$GITHUB_OUTPUT"
 
          service_account=${SERVICE_ACCOUNT}
          echo sa=$service_account >> "$GITHUB_OUTPUT"
```